src/js/react/apps/district-and-project-search/store.tsx

```diff
@@ -0,0 +1,173 @@
+import { atom } from 'jotai';
+
+import { AGGREGATIONS } from './query/queries';
+import Settings from './enum/Settings';
+import IndexFields from './enum/IndexFields';
+import useAggregations from './hooks/useAggregations';
+import type URLParams from './types/URLParams';
+import type OptionType from './types/OptionType';
+
+type configurations = {
+  error: Error|null,
+  aggs: any
+};
+
+const getParams = (searchParams: URLSearchParams) => {
+  const params: { [k: string]: any } = {};
+  const entries = searchParams.entries();
+  let result = entries.next();
+
+  while (!result.done) {
+    const [key, value] = result.value;
+
+    if (!value) {
+      result = entries.next();
+    }
+    else {
+      const existing = params[key];
+      if (existing) {
+        const updatedValue = Array.isArray(existing) ? [...existing, value] : [existing, value];
+        params[key] = updatedValue;
+      } else {
+        params[key] = [value];
+      }
+
+      result = entries.next();
+    }
+  }
+
+  return params;
+};
+
+export const urlAtom = atom<URLParams>(getParams(new URLSearchParams(window.location.search)));
+
+export const urlUpdateAtom = atom(null, (get, set, values: URLParams) => {
+  // set atom value
+  values.page = values.page || '1';
+  set(urlAtom, values);
+
+  // Set new params to window.location
+  const newUrl = new URL(window.location.toString());
+  const newParams = new URLSearchParams();
+
+  // eslint-disable-next-line
+  for (const key in values) {
+    const value = values[key as keyof URLParams];
+
+    if (Array.isArray(value)) {
+      value.forEach((option: string) => newParams.append(key, option));
+    } else if (value) {
+      newParams.set(key, value.toString());
+    } else {
+      newParams.delete(key);
+    }
+  }
+
+  newUrl.search = newParams.toString();
+  window.history.pushState({}, '', newUrl);
+});
+
+export const setPageAtom = atom(null, (get, set, page: string) => {
+  const url = get(urlAtom);
+  set(urlUpdateAtom, { ...url, page });
+});
+
+export const pageAtom = atom((get) => Number(get(urlAtom)?.page) || 1);
+
+export const configurationsAtom = atom(async(): Promise<configurations> => {
+  const proxyUrl = drupalSettings?.helfi_kymp_district_project_search.elastic_proxy_url;
+  const url: string | undefined = proxyUrl || process.env.REACT_APP_ELASTIC_URL;
+  const ndjsonHeader = '{}';
+
+  const body =
```

Comment:
With only one query used here, `_msearch` is not needed. Can use `_search` instead.